### PR TITLE
[DOC]: adding a grid to the style sheet reference.

### PR DESCRIPTION
**Pull request by kostyafarber**
## PR Summary
This PR adds gridlines to the last plot in the stylesheets example and adds an annotation `ax.grid(True)` as well as a divider to clearly distinguish between subplots. This addresses https://github.com/matplotlib/matplotlib/issues/23601
## PR Checklist

**Tests and Styling**
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

### CODE REVIEW

**Comment by kostyafarber** [comment]
Added @Kinza-Raza

File: examples/style_sheets/style_sheets_reference.py [hunk 1]
```diff
@@ -65,6 +66,12 @@ def plot_colored_circles(ax, prng, nb_samples=15):
     for sty_dict, j in zip(plt.rcParams['axes.prop_cycle'], range(nb_samples)):
         ax.add_patch(plt.Circle(prng.normal(scale=3, size=2),
                                 radius=1.0, color=sty_dict['color']))
+    ax.grid(visible=True)
+    # Add annotation for enabling grid
+    ax.annotate('ax.grid(True)', xy=(1, 1),
```
Comment (jklymak) [inline examples/style_sheets/style_sheets_reference.py:71]: This looks good, but why did you use annotate here, versus `title`?  Might also consider `family='monospace'`?

Comment (kostyafarber) [reply examples/style_sheets/style_sheets_reference.py:71]: This was added by the last author and I just incorporated those changes. 

Can refactor to use title and change the font family if desired. 

Let me know @jklymak

**Comment by QuLogic** [comment]
Please title the commit/PR correctly. You are adding a grid to the style sheet reference, not a divider and annotation. The latter is an implementation detail of the former.

**Comment by kostyafarber** [comment]
No worries will change the PR title.

**Comment by kostyafarber** [comment]
@QuLogic changed the PR title.

File: examples/style_sheets/style_sheets_reference.py (hunk 1 above)
Comment (kostyafarber) [reply examples/style_sheets/style_sheets_reference.py]: @jklymak ended up implementing your suggestions. Changed to `title` and `family='monospace'`. Fewer lines of code and more readable. Thanks.

**Comment by kostyafarber** [comment]
@jklymak let me know if I need to do anything else. Thanks!

File: examples/style_sheets/style_sheets_reference.py [hunk 2]
```diff
@@ -65,8 +66,11 @@ def plot_colored_circles(ax, prng, nb_samples=15):
     for sty_dict, j in zip(plt.rcParams['axes.prop_cycle'], range(nb_samples)):
         ax.add_patch(plt.Circle(prng.normal(scale=3, size=2),
                                 radius=1.0, color=sty_dict['color']))
-    # Force the limits to be the same across the styles (because different
-    # styles may have different numbers of available colors).
+    ax.grid(visible=True)
+
+    # Add title for enabling grid
+    plt.title('ax.grid(True)', family='monospace', y=1.075)
```
Comment (jklymak) [inline examples/style_sheets/style_sheets_reference.py:72]:
```suggestion
    plt.title('ax.grid(True)', family='monospace', fonstize='small')
```
Did you really need to set y by hand?

Comment (kostyafarber) [reply examples/style_sheets/style_sheets_reference.py]: No I suppose not. Was trying to get it looking as close to @timhoffm mockup. Thanks for the suggestion @jklymak

File: examples/style_sheets/style_sheets_reference.py [hunk 3]
```diff
@@ -65,8 +66,11 @@ def plot_colored_circles(ax, prng, nb_samples=15):
     for sty_dict, j in zip(plt.rcParams['axes.prop_cycle'], range(nb_samples)):
         ax.add_patch(plt.Circle(prng.normal(scale=3, size=2),
                                 radius=1.0, color=sty_dict['color']))
-    # Force the limits to be the same across the styles (because different
-    # styles may have different numbers of available colors).
+    ax.grid(visible=True)
+
+    # Add title for enabling grid
+    plt.title('ax.grid(True)', family='monospace', fonstize='small')
```
Comment (jklymak) [inline examples/style_sheets/style_sheets_reference.py:72]:
```suggestion
    plt.title('ax.grid(True)', family='monospace', fontsize='small')
```

Sorry if that was my typo, but check that the example runs before running CI ;-)

Comment (kostyafarber) [reply examples/style_sheets/style_sheets_reference.py:72]: No worries, I should have checked. Just ran the example it works.

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Thanks for your work and patience with this @kostyafarber .  I think this looks good, and should be helpful for folks curious about what happens if they turn the grid on.

**Pull request merged**